cmd/prometheus/main.go

```diff
@@ -166,6 +168,12 @@ func main() {
 	a.Flag("alertmanager.timeout", "Timeout for sending alerts to Alertmanager.").
 		Default("10s").SetValue(&cfg.notifierTimeout)
 
+	a.Flag("alertmanager.outage-tolerance", "Max time to tolerate prometheus outage for restoring 'for' state of alert.").
+		Default("1h").SetValue(&cfg.outageTolerance)
+
+	a.Flag("alertmanager.for-grace-period", "Minimum duration between alert and restored 'for' state. Applies only for alerts with 'for' time greater than grace period.").
```

Comment:
That's not quite right. If I have an alert with a for of 30m and it was 29m pending, then we should have to wait 10m before setting it to fire.

Reply:
Yes, it does that. Maybe the description was not clear. 

Are you referring to "Applies only for alerts with 'for' time greater than grace period"?
If yes, then I actually meant the `for` duration set in the config, and not the `for` duration remaining.